exchange/bidder.go

```diff
@@ -155,8 +166,16 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb.Bi
 	for i := 0; i < len(reqData); i++ {
 		httpInfo := <-responseChannel
 		// If this is a test bid, capture debugging info from the requests.
+		// Write debug data to ext in case if:
+		// - debugContextKey (url param) in true
+		// - account debug is allowed
+		// - bidder debug is allowed
 		if debugInfo := ctx.Value(DebugContextKey); debugInfo != nil && debugInfo.(bool) {
-			seatBid.httpCalls = append(seatBid.httpCalls, makeExt(httpInfo))
+			if accountDebugAllowed {
+				if bidder.config.DebugInfo.Allow {
```

Comment:
In case we bid more than one bidder in one request, some of those bidders may have debug.allowed=true and others may have debug.allowed=false. For this case we need to make 2 checks and write logs for those bidders where debug.allowed=true. Hope this make sense.

Reply:
Wait... did my comment disappear? 
Correct. If account level debug disabled, we don't need to check bidder level config. We just skip it. 
But if account debug is allowed, we need to check every bidder in request. 

We may bid more than one bidder in one request and those bidders may have different debug options. To handle this case we need to check each of them and log data only for those bidders where debug.allow = true. 

Hope this makes sense and answers your question.